cmd/jaeger/internal/extension/jaegerquery/server.go

```diff
@@ -54,8 +59,8 @@ func (s *server) Start(_ context.Context, host component.Host) error {
 	if err != nil {
 		return fmt.Errorf("cannot create span reader: %w", err)
 	}
-	// TODO
-	// spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, baseFactory.Namespace(metrics.NSOptions{Name: "query"}))
+
+	spanReader = storageMetrics.NewReadMetricsDecorator(spanReader, queryMetricsFactory)
```

Comment:
is reader the only moving part that gets wrapped with metrics decorator?

Reply:
Yes.